pycense/__init__.py

```diff
@@ -0,0 +1,122 @@
+import re
+import os
+import subprocess
+
+__all__ = [
+        'get_pyfiles',
+        'get_requirements',
+        'scan_file',
+        'scan_pyfile',
+        'scan_requirements',
+        'installed',
+        'get_ignored_packages'
+        ]
+
+py_imports_re  = re.compile(r'(?:from|import) ([_a-zA-Z0-9]+)(?:.*)')
+req_package_re = re.compile(r'([a-z_]+)(?:(?:[=<>~,]+)(?:[0-9]+))?')
+
+try:
+    from pip import get_installed_distributions
+except:
+    from pip._internal.utils.misc import get_installed_distributions
+
+installs = [distro.key for distro in get_installed_distributions()]
+
+exclusions = ['__pycache__', '.git']
+
+def valid_path(path):
+    return not any(e in path for e in exclusions)
+
+def installed(import_statement):
+    for install in installs:
+        if import_statement == install:
+            return True
+
+    return False
+
+def get_pyfiles(path):
+    pyfiles = []
+    for (dirpath, dirnames, filenames) in os.walk(path):
+        if valid_path(dirpath):
+            pyfiles += [os.path.join(os.path.abspath(dirpath), f) for f in filenames if f.endswith('.py')]
+    return pyfiles
+
+def get_requirements(path):
+    reqfiles = []
+    for (dirpath, dirnames, filenames) in os.walk(path):
+        if valid_path(dirpath):
+            reqfiles += [ \
```

Comment:
Feels like this could all be one line (and thus not the weird-to-see line ending escapes here)